pkg/appregistry/appregistry.go

```diff
@@ -41,6 +42,8 @@ func (f *factory) New(options Options) (Client, error) {
 	}
 
 	transport := httptransport.New(u.Host, u.Path, []string{u.Scheme})
+	transport.Transport = metrics.GetRoundTripperFunc()
```

Comment:
What happens here if this isn't initialized? Should this method we're calling here have a base case in that event?

Reply:
The init function for the `RoundTripperFunc` will succeed or cause marketplace to exit.

Reply:
Right but what about the case where we don't care about actually getting the metrics to run? (offline cluster, upstream, etc)?

Reply:
Although the situations you describe wouldn't cause an issue given that the metrics are just served on a port, you have convinced me that marketplace shouldn't exit when we fail to register the metrics.

We can ensure that marketplace continues to run if metrics fail to register by making the following changes:
1. Update the `roundTripper` variable to be set to `http.DefaultRoundTripper` by default.
2. Attempt to register the metrics. This involves changing the `prometheus.MustRegister` (which exits on failure) to `prometheus.Register` (which returns an error) and logging that error.
3. If the metrics are successfully registered, update the `roundTripper` variable to the one returned by `appRegistryRoundTripperCounter`.